render/detailed/node.go

```diff
@@ -0,0 +1,303 @@
+package detailed
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/weaveworks/scope/probe/docker"
+	"github.com/weaveworks/scope/probe/host"
+	"github.com/weaveworks/scope/probe/kubernetes"
+	"github.com/weaveworks/scope/probe/process"
+	"github.com/weaveworks/scope/render"
+	"github.com/weaveworks/scope/report"
+)
+
+// Node is the data type that's yielded to the JavaScript layer when
+// we want deep information about an individual node.
+type Node struct {
+	ID       string            `json:"id"`
+	Label    string            `json:"label"`
+	Rank     string            `json:"rank,omitempty"`
+	Pseudo   bool              `json:"pseudo,omitempty"`
+	Controls []ControlInstance `json:"controls"`
+	Metadata []MetadataRow     `json:"metadata,omitempty"`
+	Metrics  []MetricRow       `json:"metrics,omitempty"`
+	Children []ChildGroup      `json:"children,omitempty"`
+	Parents  []Parent          `json:"parents,omitempty"`
+}
+
+// ChildGroup is a topology-typed group of children for a Node.
+type ChildGroup struct {
+	Label      string         `json:"label"`
+	Nodes      []ChildSummary `json:"nodes"`
+	TopologyID string         `json:"topologyId"`
+}
+
+// ChildSummary is summary information about a child for a Node.
+type ChildSummary struct {
```

Comment:
Could this type be the same as Node?  It seems to have a lot in common, and would make generating them use the same (or some subset) of the code from MakeNode.